# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintObject.cpp
# Commit: 6fd57b83bb52
# Full Hash: 6fd57b83bb52e9683bade8cbfb5f8ba6575ac6af
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-02-07 03:55:51
# Regressor Bug: 1582832
# File Overlap Count: 3
# Description:
#   Bug 1582832: Part 1 - Make FrameLoader owner rather than DocShell responsible for discarding a BC. r=nika
#   
#   There are all sorts of lifecycle issues which arise from making DocShell
#   responsible for discarding BrowsingContexts. In this particular bug, we tend
#   to run into them in cases where we create a BrowsingContext for a FrameLoader,
# ==============================================================================

diff -r 28a9e4b075af -r 6fd57b83bb52 layout/printing/nsPrintObject.cpp
--- a/layout/printing/nsPrintObject.cpp	Thu Feb 06 18:35:10 2020 +0000
+++ b/layout/printing/nsPrintObject.cpp	Thu Feb 06 19:07:56 2020 +0000
@@ -49,10 +49,9 @@
 
   DestroyPresentation();
   if (mDidCreateDocShell && mDocShell) {
-    nsCOMPtr<nsIBaseWindow> baseWin(do_QueryInterface(mDocShell));
-    if (baseWin) {
-      baseWin->Destroy();
-    }
+    RefPtr<BrowsingContext> bc(mDocShell->GetBrowsingContext());
+    nsDocShell::Cast(mDocShell)->Destroy();
+    bc->Detach();
   }
   mDocShell = nullptr;
   mTreeOwner = nullptr;  // mTreeOwner must be released after mDocShell;